joblib/hashing.py

```diff
@@ -199,6 +199,11 @@ def save(self, obj):
                 # XXX: There might be a more efficient way of doing this
                 obj_c_contiguous = obj.flatten()
 
+            # Avoid a value error "ValueError: new type not compatible with
+            # array.", triggered when hashing array of empty shape.
+            if len(obj_c_contiguous.shape) == 0:
```

Comment:
I assumed that `obj.flags.contiguous` was `obj.flags.f_contiguous or obj.flags.c_contiguous` .  Turns out it is not the case it's just `obj.flags.contiguous`, slightly misleading ...